node/pkg/common/symmetric_test.go

```diff
@@ -1,24 +1,36 @@
 package common
 
 import (
+	"github.com/stretchr/testify/assert"
 	"testing"
 )
 
 func TestAESGCM(t *testing.T) {
-	data := []byte("cat")
-	key := []byte("01234567890123456789012345678901")
 
-	enc, err := EncryptAESGCM(data, key)
-	if err != nil {
-		t.Fatal(err)
+	type test struct {
+		data []byte
 	}
 
-	dec, err := DecryptAESGCM(enc, key)
-	if err != nil {
-		t.Fatal(err)
+	tests := []test{
+		{data: []byte("lower")},
+		{data: []byte("UPPER")},
+		{data: []byte("Mixed")},
+		{data: []byte("AlphaNum1")},
+		{data: []byte("12345")},
 	}
 
-	if string(dec) != string(data) {
-		t.Fatalf("expected %s got %s", string(data), string(dec))
+	key := []byte("01234567890123456789012345678901")
+
+	for _, tc := range tests {
+		// Test that we can encrypt data without error.
```

Comment:
This is a good refactor with 1 caveat. The way you did this golang treats this as one MONSTER test and if a single assertion fails, the rest never run. It is better to have go treat each of the test cases in `tests` be an individual test that can be run independently.

```suggestion

	for _, testCase := range tests {
		// Test that we can encrypt data without error.
		t.Run(string(testCase.data), funct(t *testing.T){
```

Note that if you do that you'll have an extra { and ( you need to close.

Reply:
Made those changes, thank you!